src/scripts/main.js

```diff
@@ -1 +1,70 @@
 'use strict';
+
+let click = false;
+let contextmenu = false;
+
+function getNotification(className, message) {
+  const notification = document.createElement('div');
+
+  notification.setAttribute('data-qa', 'notification');
+  notification.className = className;
+  notification.textContent = message;
+  document.body.append(notification);
+}
+
+const firstPromise = new Promise((resolve, reject) => {
+  addEventListener('click', () => {
+    click = true;
+
+    clearTimeout(timerID);
+    resolve('First promise was resolved');
+  });
+
+  const timerID = setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+  }, 3000);
+});
+
+const secondPromise = new Promise((resolve) => {
+  addEventListener('click', () => {
+    click = true;
+
+    resolve('Second promise was resolved');
+  });
+
+  addEventListener('contextmenu', (e) => {
+    e.preventDefault();
+
+    contextmenu = true;
+
+    resolve('Second promise was resolved');
+  });
+});
+
+const thirdPromise = new Promise((resolve) => {
+  addEventListener('click', () => {
```

Comment:
Ensure that the logic for resolving the third promise is correct. It depends on the `contextmenu` variable, which might not be set as expected if the event listeners are not managed properly.